src/lang-go.ts

```diff
@@ -246,8 +238,11 @@ async function connectAndInitialize(
             }
         )
     } catch (e) {
-        if ('message' in e && (/no such host/.test(e.message) || /i\/o timeout/.test(e.message))) {
-            console.error(connectingToSourcegraphHelp)
+        if (
+            'message' in e &&
+            (/no such host/.test(e.message) || /i\/o timeout/.test(e.message) || /connection refused/.test(e.message))
```

Comment:
Where do these errors come from? How can we be confident that these error messages are reliable? Are there no error codes/types, and if not, can we add them?

Which errors can happen that should not pass through this?